service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/HearingBookingDetailsControllerSubmittedTest.java

```diff
@@ -57,6 +57,22 @@ void shouldTriggerPopulateDatesEventWhenThereAreEmptyDates() {
 
     @Test
     void shouldNotTriggerPopulateDatesEventWhenThereAreNoEmptyDates() {
+        CallbackRequest callbackRequest = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .jurisdiction(JURISDICTION)
+                .caseTypeId(CASE_TYPE)
+                .data(Map.of("state", "Submitted"))
+                .build())
+            .build();
+
+        postSubmittedEvent(callbackRequest);
+
+        verify(coreCaseDataService, never()).triggerEvent(any(), any(), any(), any(), any());
```

Comment:
be careful with negative assertions like this, when async procession it does not prove it has not be invoked, e.g it could be 100ms later

Reply:
Have changed to use checkThat(), please check if its ok